### ISSUE TITLE
VLAN MTU not changing after install / configuring

### ISSUE THREAD

**Issue by upended** [opnsense/core#4032]
**The current OPNsense version where the bug first appeared.**
OPNsense 20.1.4-amd64 FreeBSD 11.2-RELEASE-p18-HBSD 
OpenSSL 1.1.1f 31 Mar 2020

**The last OPNsense version where the bug did not exist** 
Unknown how to extract this information. 

**The exact URL of the GUI page involved (if any)**
https://192.168.1.1/interfaces.php?if=opt1
https://192.168.1.1/interfaces.php?if=opt2
(others as well, all vlan interfaces)

[-] I have searched the existing issues and I'm convinced that mine is new.
Searched, there is a related report.  However, not in proper format and no further postings / entries for over 3 months. Ticket closed.

**Describe the bug**
**VLAN MTU is not set to value configured during installation.**

Initial steps take for installation / configuration.

1. Installed O.S. and used configuration wizard. (Original MTU for LAN was 1500.)
2. Created all VLANs needed.  
3. Attempted to change VLANs for Jumbo frames (MTU 9000), received error message indicating LAN was smaller and needed to be configured with larger value to match.  
4. Changed LAN MTU value to 9000.
5. Configured VLANs requiring MTU 9000
6. Changed VLANs priority  Interfaces> Other Types> VLAN

Value stated on webpage for LAN and VLANs modified for Jumbo frames, match MTU 9000.

Upgraded Opnsense to 20.1 (current running version)

Received error from other internal hardware communication issues.
Reviewed interfaces:
LAN MTU 9000     
VLAN 10 MTU 9000
VLAN 20 MTU 9000
VLAN 30 MTU 9000
VLAN 40 MTU 1500
........

After troubleshooting, found issue.   LAN interface did match MTU 9000; however, VLANs configured for 9000 reflecting 1500. Verified on Interfaces>Overview, MTU is set for 1500 and not 9000 as configured through webpage for interface.

**Config. xml  (configuration backup)**
  <interfaces>
    <lan>
      <if>bce0</if>
      <descr/>
      <enable>1</enable>
      <lock>1</lock>
      <spoofmac/>
      <blockbogons>1</blockbogons>
      <mtu>9000</mtu>
      <ipaddr>192.168.31.1</ipaddr>
      <subnet>24</subnet>
    </lan>
........
<opt1>
      <if>bce0_vlan10</if>
      <enable>1</enable>
      <lock>1</lock>
      <spoofmac/>
      <blockbogons>1</blockbogons>
      <mtu>9000</mtu>
    </opt1>
    <opt2>
      <if>bce0_vlan20</if>
      <enable>1</enable>
      <lock>1</lock>
      <spoofmac/>
      <blockbogons>1</blockbogons>
      <mtu>9000</mtu>
    </opt2>
    <opt3>
      <if>bce0_vlan30</if>
      <enable>1</enable>
      <lock>1</lock>
      <spoofmac/>
      <blockbogons>1</blockbogons>
      <mtu>9000</mtu>
    </opt3>

**Opnsense console**
bce0: flags=8943<UP,BROADCAST,RUNNING,PROMISC,SIMPLEX,MULTICAST> metric 0 mtu 9000
bce0_vlan10: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> metric 0 mtu 1500
bce0_vlan30: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> metric 0 mtu 1500
bce0_vlan20: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> metric 0 mtu 1500
bce0_vlan40: flags=28943<UP,BROADCAST,RUNNING,PROMISC,SIMPLEX,MULTICAST,PPROMISC> metric 0 mtu 1500

*Tip*: to validate your setup was working with the previous version, use opnsense-revert (https://docs.opnsense.org/manual/opnsense_tools.html#opnsense-revert)

**To Reproduce**
Repeat Steps take as this was a new installation and I have never used this software before.

**Expected behavior**
When initial configuration of VLANs set for MTU 9000, expection was for the interface to modify initial value (1500) and retain new value (9000).

**Screenshots**

**Relevant log files**
If applicable, information from log files supporting your claim.

**Additional context**

**Environment**
OPNsense 20.1.4-amd64 FreeBSD 11.2-RELEASE-p18-HBSD 
OpenSSL 1.1.1f 31 Mar 2020
Chipset : Intel C202
NIC: Broadcom 5709

![Overview-Int LAN](https://user-images.githubusercontent.com/63566363/79084884-29488a00-7ceb-11ea-8005-5d1c0b46ba66.JPG)
![Overview-Int Opt2](https://user-images.githubusercontent.com/63566363/79084886-2c437a80-7ceb-11ea-91b2-c005af363ffb.JPG)
![Overview-Int Opt3](https://user-images.githubusercontent.com/63566363/79084888-2ea5d480-7ceb-11ea-8529-3de3a4ce7fc5.JPG)

**Comment by Uglymotha** [comment]
Having the same issue here. Everywhere I can configure MTU it's set to 9000, vlan interfaces do not get configured correctly on physical interfaces. They do get configured correctly on LACP trunk:
[root@rdifw01 /conf]# grep  "<mtu>" config.xml
      <mtu>9000</mtu>
      <mtu>9000</mtu>
      <mtu>9000</mtu>
      <mtu>9000</mtu>
      <mtu>9000</mtu>
      <mtu>9000</mtu>
      <mtu>9000</mtu>
      <mtu>9000</mtu>
      <mtu>9000</mtu>
      <mtu>9000</mtu>
      <mtu>9000</mtu>
      <mtu>9000</mtu>
      <mtu>9000</mtu>
      <mtu>9000</mtu>
      <mtu>9000</mtu>
      <mtu>9000</mtu>

[root@rdifw01 /conf]# ifconfig lagg0
lagg0: flags=8943<UP,BROADCAST,RUNNING,PROMISC,SIMPLEX,MULTICAST> metric 0 mtu 9000

[root@rdifw01 /conf]# ifconfig lagg0_vlan100
lagg0_vlan100: flags=8943<UP,BROADCAST,RUNNING,PROMISC,SIMPLEX,MULTICAST> metric 0 mtu 9000

[root@rdifw01 /conf]# ifconfig igb2
igb2: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> metric 0 mtu 9000

[root@rdifw01 /conf]# ifconfig igb2_vlan100
igb2_vlan100: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> metric 0 mtu 1500

**Comment by Uglymotha** [comment]
Verified workaround for this:
1. Create a LAGG, set protocol to Failover, add the physical interface
2. Create VLANs on the LAGG
3. Assign the VLAN interface and configure IP

I actually see some flexibility benefits in going this route, so I'm sticking with it.

**Comment by Uglymotha** [comment]
Another minor issue, lowering MTU of vlan interface created as above only takes effect after reboot.

**Comment by AdSchellevis** [comment]
This issue has been automatically timed-out (after 180 days of inactivity).

For more information about the policies for this repository,
please read https://github.com/opnsense/core/blob/master/CONTRIBUTING.md for further details.

If someone wants to step up and work on this issue,
just let us know, so we can reopen the issue and assign an owner to it.

**Comment by RedVortex** [comment]
Same issue here. Interface (ix1) MTU is configured at 9000. Vlan (ix1_vlan420) is also configured at 9000.

After saving, applying and even rebooting. The interface is in fact 9000 in overview. But the vlan is only 1500.

This seems to happen with the interface and the vlan interface have the same MTU. When that happens, it looks like the vlan default to 1500, instead of what is configured.

If I put let's say 9000 on ix1 and 8999 on ix1_vlan420. 9000 is now applied properly to the vlan, I see 8999 on the vlan now. If I put 9000 again on the vlan, the vlan now goes back 1500.

I'm running 21.1.4 (latest at the time of this post).

**Comment by RedVortex** [comment]
I also upgraded to 21.1.5 and I confirm the issue is still present in this version.

@AdSchellevis I think this issue should be re-opened, this is easily reproducible and it is most likely a logic or network interface config file issue when the vlan and the interface MTU match, it seems like the vlan MTU configures itself for 1500 instead of what we configure in the UI.

**Comment by RedVortex** [comment]
I'm not sure any of us can code to help you out to fix this but it is definitely a bug in opnsense, affecting all free and commercial users as well. Is there a better way to open up a bug with opnsense and get this fixed besides here ? Also, I wonder why a bug would be closed when it is reproducible and not fixed ?

**Comment by AdSchellevis** [comment]
best read the contributor guidelines https://github.com/opnsense/core/blob/master/CONTRIBUTING.md, if there's new / other information which might shed another light on the subject (either if it's a but in OPNsense, an upstream omission or just a configuration issue), I don't mind if someone opens a new ticket. Dragging tickets endlessly doesn't really help in our opinion, which is why we have implemented policies to handle them.

**Comment by upended** [comment]
This issue is significant and a programming error. I have attempted to use the workaround.  However, once all the physical interface has been assigned, it can not be used as there is no additional physical interfaces as LAGG requires.  

Can the ticket be re-opened and label be changed from support to bug as well?

When reading the guidelines, it states 

> It is often better to join an ongoing discussions on similar open issues than creating a new one as there may be workarounds or ideas available.
...

> As a general rule, closed tickets shall and will not be responded to.

If this ticket is to remained closed, it will most likely not be reviewed, nor fixed and it will become lost over time even though it is an "open issue".  Keeping this ticket open will allow anyone who has the technical capability to see it as an outstanding issue and correct it.  Also, since the guidelines state there will be no response, this give more reason to keeping this ticket open.

**Comment by fichtner** [comment]
Please just adhere to contribution guidelines. A problem that can’t be resolved can’t be tracked forever in any case?

**Comment by lukvdborne** [comment]
I have the same problem on 21.7.0. The NIC is an "Intel(R) I211"

Here are the steps to reproduce:
- Set mtu=9000 on a LAN interface (in my case, igb1)
- Check Interfaces > Overview to see the mtu is indeed 9000
- Create a VLAN in Interfaces > Other Types > VLAN with the physical LAN interface as parent
- Create interface assignment in Interfaces > Assignments for the VLAN you just created
- Go to igb1_vlan10 (or whichever interface you just created) in Interfaces, set an mtu of 9000 and save and apply changes
- Go to Interfaces > Overview and see that the mtu of the VLAN interface is set to 1500. Expected value: 9000
- If you go back to Interfaces > [igb1_vlan10] you'll see that the interface did at least save the value of 9000 correctly. 
- Just to be sure, login to a shell and conclude that ifconfig too states that the mtu is still 9000. 

Hope this helps.

**Comment by upended** [comment]
To those that are experiencing this issue (assuming it has not been fixed yet), the simplest way to change this is to 

**BEFORE STARTING EXPORT A BACKUP OF YOUR CONFIGURATION FILE (keep unaltered)**

1. Export the configuration file.  
2. Go to Interfaces:Overview find your vlan and notate the OPT#.  
3. Open it in a text editor, looking for OPT#.
4. Change the MTU value.
5. Upload the configuration.
6. Reboot

or 

(This is not even finished, very complete relative to the first option)
1. Log into G.U.I.
2. Go to Interfaces:Overview   (Lookup opt# for the V.L.A.N. you want to change the M.T.U. on.)
3. Log into C.L.I. through S.S.H.  
4. > cat /conf/config.xml | grep "<opt#>"
         example: cat /conf/config.xml | grep "<opt12>"     (Look at the line number at the right)
5. > vi /conf/config.xml
6. press the ESC key
7. :
          The above is a colon, enter it as you would in any document
8. set number
          To the right of the colon prompt enter above
9. press enter key
10. go to line needed and change the value.
11. press 

Note: interface:XXXX is not able to retain settings, but also can not reflect the changes being made in the C.L.I. event after saving / refreshing. Changes will show on Interfaces:Overview

**Comment by omenos** [comment]
Came across this bug while messing around with OPNsense today in a VM. This issue doesn't seem specific to jumbo frames, but actually setting a VLAN MTU to the same as its parent interface. For example:

If I set LAN (vtnet0) to 9000, it shows correcting in the interface config, Interfaces > Overview, and Interfaces > Diagnostics > Netstat > Interfaces, and `ifconfig vtnet0`. If I then set vtnet0_vlan10 to 9000, it will show 9000 in the interface config, but no where else in the OPNsense GUI or command line. Setting it to anything <9000 works just fine.

If I then drop the VLAN back to 1500 and then the parent interface as well, everything works fine. If I then drop the VLAN to <1500 and back, it will do the same thing, and prevent the VLAN from hitting 1500, staying at the previous lower value. But it will still report the 'new' 1500 value in the VLAN config. The only way to rectify this is to either

a) raise the parent, then VLAN, then drop parent or
b) `ifconfig <if> mtu <desired_value>` or
c) modify the configs directly as others have mentioned.

Performing 'b' will show correctly in the Interfaces > {Overview, Diagnostics > Netstat} while not changing the interface config value.

**Comment by RedVortex** [comment]
This issue persists in 22.1, just re-checked it.

**Comment by fichtner** [comment]
@RedVortex would you mind opening a new ticket for me with steps to reproduce. Happy to look at it in an isolated report now that dhcp(v4) is back where it should be. 😊


Cheers,
Franco

**Comment by RedVortex** [comment]
> @RedVortex would you mind opening a new ticket for me with steps to reproduce. Happy to look at it in an isolated report now that dhcp(v4) is back where it should be. 😊

hehehe totally agree, done !

**Comment by fichtner** [comment]
A promising patch for 22.1 was submitted here https://github.com/opnsense/core/issues/5546#issuecomment-1031278519 -- all help in confirming it works are greatly appreciated.


Cheers,
Franco
